components/service-operator/controllers/image_repository_cloudformation_test.go

```diff
@@ -0,0 +1,159 @@
+package controllers_test
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	access "github.com/alphagov/gsp/components/service-operator/apis/access/v1beta1"
+	storage "github.com/alphagov/gsp/components/service-operator/apis/storage/v1beta1"
+	"github.com/alphagov/gsp/components/service-operator/internal/aws/cloudformation"
+	"github.com/alphagov/gsp/components/service-operator/internal/object"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	core "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var _ = Describe("ImageRepositoryCloudFormationController", func() {
+
+	var timeout time.Duration = time.Minute * 15
+	var client client.Client
+	var ctx context.Context = context.Background()
+	var teardown func()
+
+	BeforeEach(func() {
+		client, teardown = SetupControllerEnv()
+	})
+
+	AfterEach(func() {
+		teardown()
+	})
+
+	It("Should create and destroy an ECR repository", func() {
+
+		var (
+			name                   = fmt.Sprintf("test-image-%s", time.Now().Format("20060102150405"))
+			secretName             = "test-ecr-secret"
+			principalName          = "test-role"
+			namespace              = "test"
+			resourceNamespacedName = types.NamespacedName{
+				Namespace: namespace,
+				Name:      name,
+			}
+			secretNamespacedName = types.NamespacedName{
+				Namespace: namespace,
+				Name:      secretName,
+			}
+			principal = access.Principal{
+				TypeMeta: metav1.TypeMeta{
+					APIVersion: access.GroupVersion.Group,
+					Kind:       "Principal",
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: namespace,
+					Name:      principalName,
+					Labels: map[string]string{
+						cloudformation.AccessGroupLabel: "test.access.group",
+					},
+				},
+			}
+			repository = storage.ImageRepository{
+				TypeMeta: metav1.TypeMeta{
+					Kind:       "ImageRepository",
+					APIVersion: storage.GroupVersion.Group,
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Name:         name,
+					GenerateName: "",
```

Comment:
Does this do anything?

Reply:
"" is the zero value, so this is just the same as it not being there